### PR TITLE
Made changes to create the COS secret from the template

### PR DESCRIPTION
**Pull request by Shirisha-S-Rao-1**

### CODE REVIEW

**Review by akgunjal — APPROVED** [review]
LGTM

File: config-templates/redhat/ocs-local/4.6/README.md [hunk 1]
```diff
@@ -12,36 +12,11 @@ In order to deploy OCS, the following prerequisites are required.
...
-    ```
-    oc create -f os-namespace.yaml
-    ```
-
-3. Verify that the namespace is created.
-    ```
-    oc get namespaces | grep storage
-    ```
-
+- You must provision an instance of IBM Cloud Object Storage and provide your COS HMAC credentials, the regional public endpoint and the IBM COS location while creating the storage configuration.
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/README.md:15]:
```suggestion
- You must provision an instance of IBM Cloud Object Storage and provide your COS HMAC credentials, the regional public endpoint, and the IBM COS location when you create your storage configuration.
```

File: config-templates/redhat/ocs-local/4.6/README.md [hunk 2]
```diff
@@ -12,36 +12,11 @@ In order to deploy OCS, the following prerequisites are required.
...
 
 ### Creating the IBM COS service instance
 
-Run the following commands to create a COS instance, create a set of HMAC credentials, and create a Kubernetes secret that uses your COS HMAC credentials.
+Run the following commands to create a COS instance and to create a set of HMAC credentials.
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/README.md:19]:
```suggestion
Run the following commands to create a COS instance and create a set of HMAC credentials.
```

File: config-templates/redhat/ocs-local/4.6/README.md [hunk 3]
```diff
@@ -53,15 +28,11 @@ Run the following commands to create a COS instance, create a set of HMAC creden
     $ ibmcloud resource service-key-create cos-cred-rw Writer --instance-name noobaa-stor --parameters '{ "HMAC": true}'
     ```
 
-3. Create the Kubernetes secret named `ibm-cloud-cos-creds` in the `openshift-storage` namespace that uses your COS HMAC credentials. When you run the command, specify your COS HMAC access key ID and secret access key.
+3. Get the regional public endpoint and the IBM COS location from the UI (Select the region closest to your cluster) or by running this command
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/README.md:31]:
```suggestion
3. Get the regional public endpoint and the location of your IBM COS instance. You can get location & endpoint details from the [IBM Cloud consle](cloud.ibm.com) or by running the following command.
```

File: config-templates/redhat/ocs-local/4.6/README.md [hunk 4]
```diff
@@ -53,15 +28,11 @@ Run the following commands to create a COS instance, create a set of HMAC creden
...
     ```
-    $ kubectl -n 'openshift-storage' create secret generic 'ibm-cloud-cos-creds' --type=Opaque --from-literal=IBM_COS_ACCESS_KEY_ID=<access_key_id> --from-literal=IBM_COS_SECRET_ACCESS_KEY=<secret_access_key>
+    $ curl https://control.cloud-object-storage.cloud.ibm.com/v2/endpoints | jq '."service-endpoints"."regional"'
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/README.md:33]: is there another command that we can give that doesn't require jq? maybe jsonpath instead

```suggestion
    $ curl https://control.cloud-object-storage.cloud.ibm.com/v2/endpoints | jq '."service-endpoints"."regional"'
```

File: config-templates/redhat/ocs-local/4.6/README.md [hunk 5]
```diff
@@ -129,9 +100,13 @@ When you create your OCS configuration, you must specify device paths for the ob
...
 | `osd-device-path` | Required | Enter the `disk-by-id` paths to the devices that you want to use for the OSD pods. Example: `/dev/scsi-3600605b00d87b43027b3bc310a64c6c9-part2`. | N/A | csv |
 | `num-of-osd` | Optional | Enter the number of OSDs. OCS will create 3x number of OSDs for the value specified. Initial storage capacity is the same as your disk size specified at `osd-device-path`. When you want to increase your storage capacity, you have to increase `num-of-osd` by the number of disks you add (taking into consideration the replication factor, which is `3` by default) | 1 | integer |
-|`worker-nodes` | Required |Workers which need to be a part of OCS (Minimum 3). | N/A |csv |
+|`worker-nodes` | Optional |Workers which need to be a part of OCS (Minimum 3). | N/A |csv |
 | `billing-type` | Optional | Enter the billing option that you want to use. You can enter either `hourly` or `monthly`. | `hourly` | string |
 | `ocs-upgrade` | Optional | Set to `true` if you want to upgrade the major version of OCS while creating a configuration of the newer version. | false | boolean |
+| `ibm-cos-endpoint` | Required | Enter the IBM COS regional public endpoint | N/A | string |
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/README.md:106]:
```suggestion
| `ibm-cos-endpoint` | Required | Enter the IBM COS regional public endpoint. | N/A | string |
```

File: config-templates/redhat/ocs-local/4.6/README.md [hunk 6]
```diff
@@ -129,9 +100,13 @@ When you create your OCS configuration, you must specify device paths for the ob
...
+| `ibm-cos-location` | Required | Enter the IBM COS regional location | N/A | string |
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/README.md:107]:
```suggestion
| `ibm-cos-location` | Required | Enter the IBM COS regional location. Example: `https://s3.us-east.cloud-object-storage.appdomain.cloud` | N/A | string |
```

File: config-templates/redhat/ocs-local/4.6/README.md [hunk 7]
```diff
@@ -129,9 +100,13 @@ When you create your OCS configuration, you must specify device paths for the ob
...
+| `ibm-cos-access-key` | Required | Enter the IBM COS access key ID for using COS as backend storage | N/A | string |
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/README.md:108]:
```suggestion
| `ibm-cos-access-key` | Required | Enter the IBM COS access key ID. | N/A | string |
```

File: config-templates/redhat/ocs-local/4.6/README.md [hunk 8]
```diff
@@ -129,9 +100,13 @@ When you create your OCS configuration, you must specify device paths for the ob
...
+| `ibm-cos-secret-key` | Required | Enter the IBM COS secret access key for using COS as backend storage | N/A | string |
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/README.md:109]:
```suggestion
| `ibm-cos-secret-key` | Required | Enter your IBM COS secret access key. | N/A | string |
```

File: config-templates/redhat/ocs-local/4.6/custom-parameters.json [hunk 9]
```diff
@@ -47,5 +47,31 @@
     "name": "billing-type",
     "default": "hourly",
     "required": "false"
+  },
+  {
+    "description": "IBM COS regional public endpoint",
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/custom-parameters.json:52]:
```suggestion
    "description": "The IBM COS public regional endpoint.",
```

File: config-templates/redhat/ocs-local/4.6/custom-parameters.json [hunk 10]
```diff
@@ -47,5 +47,31 @@
...
+    "displayname": "IBM COS endpoint",
+    "name": "ibm-cos-endpoint",
+    "required": "true"
+  },
+  {
+    "description": "IBM COS regional location",
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/custom-parameters.json:58]:
```suggestion
    "description": "The IBM COS regional location.",
```

File: config-templates/redhat/ocs-local/4.6/custom-parameters.json [hunk 11]
```diff
@@ -47,5 +47,31 @@
...
+    "displayname": "IBM COS location",
+    "name": "ibm-cos-location",
+    "required": "true"
+  },
+  {
+    "description": "The IBM COS access key ID for using COS as backend storage",
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/custom-parameters.json:64]:
```suggestion
    "description": "The IBM COS access key ID.",
```

File: config-templates/redhat/ocs-local/4.6/custom-parameters.json [hunk 12]
```diff
@@ -47,5 +47,31 @@
...
+    "displayname": "IBM COS access key ID",
+    "name": "ibm-cos-access-key",
+    "category": "secret",
+    "required": "true"
+  },
+  {
+    "description": "The IBM COS secret access key for using COS as backend storage",
```
Comment (derekpoindexter) [inline config-templates/redhat/ocs-local/4.6/custom-parameters.json:71]:
```suggestion
    "description": "The IBM COS secret access key.",
```

**Review by derekpoindexter — CHANGES_REQUESTED** [review]
lgtm - but we need a command that doesn't require jq or we can just point to the ui to retrieve the values

File: config-templates/redhat/ocs-local/4.6/README.md (hunk 6 above)
Comment (derekpoindexter) [reply config-templates/redhat/ocs-local/4.6/README.md:107]: Also, for this location parameter ^ i dont see `us-east-standard` east in the UI, docs, or in the command results - where do they retrieve this value?

File: config-templates/redhat/ocs-local/4.6/README.md (hunk 4 above)
Comment (nkkashyap) [reply config-templates/redhat/ocs-local/4.6/README.md:33]: @Shirisha-S-Rao-1 @derekpoindexter 
Better point to docs https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-endpoints
https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-classes

**Review by derekpoindexter — APPROVED** [review]

**Pull request merged**